plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -417,6 +417,32 @@ public MongoCursor<Document> execute(MongoTableHandle tableHandle, List<MongoCol
         return iterable.iterator();
     }
 
+    static Document buildProjection(MongoTableHandle table, List<MongoColumnHandle> columns)
+    {
+        if (table.getType() == NAMED) {
+            Document output = new Document();
+            for (MongoColumnHandle column : columns) {
+                output.append(column.getName(), 1);
+            }
+            return output;
+        }
+        if (table.getType() == QUERY) {
+            return ALL_PROJECTION;
+        }
+        throw new IllegalArgumentException();
+    }
```

Comment:
Since we pass the filter query in the PTF, can we reuse the projection operation for the collection/table ?